src/AutorestSwift/Code Generator/Manager.swift

```diff
@@ -54,17 +66,32 @@ class Manager {
         }
     }
 
+    /// Initialize with a raw YAML string (i.e. the way it is received from Autorest).
+    init(withString string: String) {
+        self.mode = .autorest
+        self.inputString = Manager.sanitize(yaml: string)
+        self.destinationRootUrl = URL(fileURLWithPath: NSTemporaryDirectory(), isDirectory: true)
```

Comment:
Creates a temporary destination URL for generating code. This way we can run our linters and such on this and then copy it all using Autorest WriteFile commands.